### PR TITLE
 feat(docs): add nodeadm wasm targets to call from docs

### PR DESCRIPTION
**Pull request by ndbaker1**
**Issue #, if available:**

**Description of changes:**

It would be helpful to be able to test a `NodeConfig` straight from the browser without installing any tools. 

This PR adds a WebAssembly package containing nodeadm's config parsing hook and provides a minimal editor that users can paste their config into.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

deployed to my fork's ghpages: https://ndbaker1.github.io/amazon-eks-ami/nodeadm/doc/playground/

```shell
rm -rf ./site
pushd nodeadm && make wasm && popd
mkdir -p ./site/assets/javascripts && cp "$(go env GOROOT)/misc/wasm/wasm_exec.js" ./site/assets/javascripts/
mkdir -p site/assets/wasm && cp nodeadm/_bin/nodeadm.wasm ./site/assets/wasm
mkdocs gh-deploy -r ndbaker1 --dirty --no-history --force
```

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

File: nodeadm/wasm/assets/wasm_exec.js
Comment (cartermckinnon) [inline nodeadm/wasm/assets/wasm_exec.js:1]: Is this served from somewhere reputable (go.dev, etc) that we can pull during the build? or is it better to have it in VCS?

File: nodeadm/wasm/assets/index.html
Comment (cartermckinnon) [inline nodeadm/wasm/assets/index.html:1]: This is a very cool piece of code 🤓

File: nodeadm/wasm/assets/main.wasm
Comment (cartermckinnon) [inline nodeadm/wasm/assets/main.wasm:1]: This wouldn't be on `main`, right? We'd build it and push to `gh-pages` with the docs?

File: nodeadm/wasm/check/main.go
Comment (cartermckinnon) [inline nodeadm/wasm/check/main.go:1]: Can you move this to `cmd/nodeadm-wasm` and make `main` a bit more generic? like a `map[string]exportedFunc`

**Review by cartermckinnon — COMMENTED** [review]
💯

File: nodeadm/wasm/assets/main.wasm
Comment (ndbaker1) [reply nodeadm/wasm/assets/main.wasm:1]: hah yea oops, i should've added `*.wasm` to the gitignore 😉

File: nodeadm/wasm/assets/wasm_exec.js
Comment (ndbaker1) [reply nodeadm/wasm/assets/wasm_exec.js:1]: i don't see any great guidance here, but guessing its probably more stable to stage it from the go version used during build. in theory we could just point to https://github.com/golang/go/blob/master/lib/wasm/wasm_exec.js if thats stable

**Comment by ndbaker1** [comment]
im somewhat happy with the functionality now :)  
feel free to take a longer look @cartermckinnon

File: .github/workflows/deploy-docs.yaml [hunk 1]
```diff
@@ -10,6 +10,14 @@ jobs:
       contents: write
     runs-on: ubuntu-latest
     steps:
+    - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # 5.1.0
+    - run: |
+        pushd nodeadm
+        GOOS=js GOARCH=wasm go build -o nodeadm.wasm cmd/nodeadm-wasm/main.go
+        popd
+        mkdir -p ./site/assets/wasm && cp ./nodeadm/nodeadm.wasm ./site/assets/wasm/
+        mkdir -p ./site/assets/javascripts && cp "$(go env GOROOT)/misc/wasm/wasm_exec.js" ./site/assets/javascripts/
```
Comment (cartermckinnon) [inline .github/workflows/deploy-docs.yaml:19]: Pulling `wasm_exec.js` from the go installation is great!

File: .github/workflows/deploy-docs.yaml [hunk 2]
```diff
@@ -10,6 +10,14 @@ jobs:
       contents: write
     runs-on: ubuntu-latest
     steps:
+    - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # 5.1.0
+    - run: |
+        pushd nodeadm
+        GOOS=js GOARCH=wasm go build -o nodeadm.wasm cmd/nodeadm-wasm/main.go
```
Comment (cartermckinnon) [inline .github/workflows/deploy-docs.yaml:16]: Can you move this into a target of `nodeadm/Makefile` so it's easier to run locally and is covered by the PR checks?

**Review by cartermckinnon — COMMENTED** [review]
One small thing!

File: .github/workflows/deploy-docs.yaml (hunk 2 above)
Comment (ndbaker1) [reply .github/workflows/deploy-docs.yaml]: added `make wasm` target and `nodeadm-build-wasm` workflow 👍

**Review by cartermckinnon — APPROVED** [review]

File: .github/workflows/deploy-docs.yaml (hunk 1 above)
Comment (cartermckinnon) [reply .github/workflows/deploy-docs.yaml:17]: @ndbaker1 this failed: https://github.com/awslabs/amazon-eks-ami/actions/runs/12270078234/job/34234724232

You'll need to move this step after the `actions/checkout` one, that's where the code comes from

Comment (ndbaker1) [reply .github/workflows/deploy-docs.yaml:17]: aargh! how did i miss that one 😠 lets wrap this up: https://github.com/awslabs/amazon-eks-ami/pull/2088

**Pull request merged**